okta/okta.go

```diff
@@ -304,18 +304,22 @@ func (m *userManager) GetLoginHandler(_ string) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		nonce, err := util.RandomString()
 		if err != nil {
-			err = errors.Wrap(err, "could not get login handler")
+			err = errors.Wrap(err, "generating nonce for Okta request")
 			grip.Critical(message.WrapError(err, message.Fields{
-				"reason": "nonce could not be generated",
+				"reason":  "nonce could not be generated",
+				"context": "Okta",
+				"request": gimlet.GetRequestID(r.Context()),
 			}))
 			writeError(w, err)
 			return
 		}
 		state, err := util.RandomString()
 		if err != nil {
-			err = errors.Wrap(err, "could not get login handler")
+			err = errors.Wrap(err, "generating state for Okta request")
 			grip.Critical(message.WrapError(err, message.Fields{
-				"reason": "state could not be generated",
+				"reason":  "state could not be generated",
+				"context": "Okta",
+				"request": gimlet.GetRequestID(r.Context()),
```

Comment:
Do the request contexts hold any sensitive information that we may not want to log to splunk?

Reply:
This is just the request ID (i.e. the incrementing reqeuest number), which isn't sensitive. The error doesn't actually include the random number and it wouldn't be valid anyways since it failed to generate.